cglib/src/main/java/net/sf/cglib/proxy/Enhancer.java

```diff
@@ -413,10 +394,58 @@ private void validate() {
 
     static class EnhancerFactoryData {
         public final Class generatedClass;
-        public volatile Factory factory;
+        private final Method setThreadCallbacks;
+        private final Class[] primaryConstructorArgTypes;
+        private final Constructor primaryConstructor;
 
-        public EnhancerFactoryData(Class generatedClass) {
+        public EnhancerFactoryData(Class generatedClass, Class[] primaryConstructorArgTypes, boolean classOnly) {
             this.generatedClass = generatedClass;
+            try {
+                setThreadCallbacks = getCallbacksSetter(generatedClass, SET_THREAD_CALLBACKS_NAME);
+                setThreadCallbacks.setAccessible(true);
```

Comment:
Does it hurt? I just thought it would enable some flexibility of the method declaration.